src/StyleSheet.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { canUseDOM } from 'exenv';
+
+import { makeStyleTag, sheetForTag, injectStyles, insertStyleTag } from './utils';
+
+export const SHEET_ID = 'react-scrolllock-stylesheet';
+type Props = { styles: string };
+
+export default class Sheet extends PureComponent<Props> {
+  sheet: CSSStyleSheet | null;
+  componentDidMount() {
+    if (!canUseDOM) return;
+
+    // ensure there's only ever ONE stylesheet for react-scrolllock
+    const existingLock = document.querySelector(`#${SHEET_ID}`);
+    if (existingLock) return;
```

Comment:
It would be much _stable_ to add and remove stylesheet separately for each lock.
Having this code, and 2 locks mounted in one time - it will unlock scroll on the first lock unmount.

Meanwhile, having separated styles will work well for any number of locks and locking order.

Reply:
I tend to agree that this would be a safer algo. I don't think it should suffer from perf issues because I don't think there's a use case for mounting a scroll lock component thousands of times.

Reply:
As mentioned in [the comment above](https://github.com/jossmac/react-scrolllock/pull/28#issuecomment-394652670), the stylesheet is only removed when the last lock unmounts.

None-the-less I've updated the PR.